compiler/qsc_eval/src/lib.rs

```diff
@@ -381,51 +394,12 @@ struct Scope {
     frame_id: usize,
 }
 
-impl Default for Env {
-    #[must_use]
-    fn default() -> Self {
-        Self(vec![Scope::default()])
-    }
-}
-
-enum Cont {
-    Action,
-    Expr(ExprId),
-    Frame(usize),
-    Scope,
-    Stmt(StmtId),
-}
-
-#[derive(Clone)]
-enum Action {
-    Array(usize),
-    ArrayRepeat(Span),
-    ArrayAppendInPlace(ExprId),
-    Assign(ExprId),
-    Bind(PatId, Mutability),
-    BinOp(BinOp, Span, Option<ExprId>),
-    Call(Span, Span),
-    Consume,
-    Fail(Span),
-    Field(Field),
-    If(ExprId, Option<ExprId>),
-    Index(Span),
-    Range(bool, bool, bool),
-    Return,
-    StringConcat(usize),
-    StringLit(Rc<str>),
-    UpdateIndex(Span),
-    UpdateIndexInPlace(ExprId, Span),
-    Tuple(usize),
-    UnOp(UnOp),
-    UpdateField(Field),
-    While(ExprId, BlockId),
-}
-
 pub struct State {
-    cont_stack: Vec<Cont>,
-    action_stack: Vec<Action>,
-    vals: Vec<Value>,
+    exec_graph_stack: Vec<Rc<[ExecGraphNode]>>,
+    idx: u32,
+    idx_stack: Vec<u32>,
+    curr_val: Option<Value>,
```

Comment:
From our conversation about this PR yesterday, this seems like what we referred to as the "result register", is this correct?

If that's the case, could we consider renaming this to `val_register`? It's a small change, but I think it would help map the way we talk about the evaluator with the code itself.

Reply:
Yeah, I think that's reasonable. I can change the variable name and the accessor methods too, just make sure they are consistent.